tensorflow/lite/micro/kernels/arc_mli/mli_interface_mli_20.cc

```diff
@@ -138,9 +139,12 @@ void MliTensorInterface::SetScalePerChannel(float* fscale,
   }
 
   for (int i = 0; i < num_channels; i++) {
-    int16_t iscale =
-        (int16_t)((1ll << (*this->ScaleFracBits<int8_t**>())[i]) * fscale[i] +
-                  0.5f);
+    int32_t iscale_acc = (int32_t)((1ll << (*this->ScaleFracBits<int8_t**>())[i]) * fscale[i] + 0.5f);
+    if (iscale_acc > SHRT_MAX) {
+      (*this->ScaleFracBits<int8_t**>())[i] -= 1;
+      iscale_acc = (int32_t)((1ll << (*this->ScaleFracBits<int8_t**>())[i]) * fscale[i] + 0.5f);

```

Comment:
Having already quantized value in the iscale_acc, maybe the following produce the same result:

`iscale_acc = (iscale_acc+1) >> 1 //  shift with rounding up`